juniorguru_chick/lib/thread_names.py

```diff
@@ -0,0 +1,37 @@
+import re
+
+def name_thread(message: str) -> str | None:
+    """Returns a name for a thread created from a given message. Example messages include: [eslint, nextjs], [eslint,nextjs], [ Java ], [:css:], [:css: CSS], [eslint,nextjs, :css: CSS]. Expected output includes: eslint, nextjs, eslint,nextjs, Java, css, CSS, eslint,nextjs, CSS."""
+
+    """
+    1. Check if the message includes text in brackets, if yes:
+        a. Remove brackets
+        b. Remove spaces
+        c. Check if the message includes text between colons
```

Comment:
I think there will never be text between colons just like that. User types it there like `:css:`, but Discord translates it into either:

- Custom emoji, and it becomes `<:css:842343369618751519>`
- Unicode emoji, and it becomes `🐍`

For simplicity, I'd keep them as they are. Treat them as if they're any other words, and if that becomes a problem, let's fix that in the future. In such case, I think it should be enough to:

- See if the beginning (leading whitespace not breaking the algorithm) of the message has a match for `[`, followed by multiple "anything else than `]`", followed by `]`.
- If yes, take the "anything else than `]`", split it by commas, strip trailing white space.
- Join the resulting list back to a string and turn it into a thread name as you like.
- Keep the original message as is, we're just naming threads.
- If there are no brackets at the beginning, name the thread by the current weekday.

Potentially useful links:

- https://regex101.com/
- `str.lstrip()`, `re.match()`, `str.strip()`, `str.join()`
- Python's [verbose regular expressions](https://docs.python.org/3/library/re.html#re.VERBOSE) allow for less headaches, see e.g. [this](https://github.com/honzajavorek/junior.guru/blob/e8688cdb266d11353841343aa856ec1b49b5d9e2/juniorguru/lib/coupons.py#L4)

Also, the dispute whether to add [walrus operator](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions) to Python or not did cause Guido to step down as a BDFL, but in the end we have it! Checking results of `re.match` and `re.search` can be one line shorter now: `if match := re.match(...):` Ofc, I understand it's not for everyone, so up to you. (Just for those brave enough to roll on the beach with walruses, muhaha).